modules/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/CodeAnalyzer.java

```diff
@@ -298,8 +302,15 @@ public void visit(BLangAbort abortNode) {
 
     @Override
     public void visit(BLangRetry retryNode) {
-        if (this.failedBlockCount == 0) {
-            this.dlog.error(retryNode.pos, DiagnosticCode.RETRY_CANNOT_BE_OUTSIDE_TRANSACTION_FAILED_BLOCK);
+        boolean error = true;
```

Comment:
Can be shortened as "boolean error = !this.retryStmtCheckStack.isEmpty() && this.retryStmtCheckStack.peek();"

Reply:
Fixed with f6fe91e29e529078b0c173726ea698ee63e9fe62. Thanks.